custom_components/localtuya/cover.py

```diff
@@ -186,19 +220,194 @@ def __init__(
+    #     self._set_position_dps = set_position_dps
+    #     self._last_movement_dps = last_movement_dps
+        
+    #     print(
+    #         "Initialized tuya cover [{}] with cover status [{}] and state [{}]".format(
+    #             self._name, self._status, self._state
             )
         )
 
+    @property
+    def device_info(self):
+        return {
+            "identifiers": {
+                # Serial numbers are unique identifiers within a specific domain
+                ("LocalTuya", f"local_{self._device.unique_id}")
+            },
+            "name": self._device._friendly_name,
+            "manufacturer": "Tuya Generic",
+            "model": "SmartCover",
+            "sw_version": "3.3",
+        }
+
+    @property
+    def open_cmd(self):
+        """Get name of open command."""
+        _LOGGER.info("def open_cmd called=%s", self._open_cmd)
+        return self._open_cmd
+
+    @property
+    def close_cmd(self):
+        """Get name of close command."""
+        _LOGGER.info("def close_cmd called=%s", self._close_cmd)
+        return self._close_cmd
+
+    @property
+    def stop_cmd(self):
+        """Get name of stop command."""
+        _LOGGER.info("def close_cmd called=%s", self._stop_cmd)
+        return self._stop_cmd
+
+    @property
+    def last_movement_dps(self):
+        """Get last movement key"""
+        #_LOGGER.info("def last_movement_dps called=%s", self._last_movement_dps)
+        return self._last_movement_dps
+
+
+    @property
+    def last_movement(self):
+        #_LOGGER.info("def last_movement called=%s", self._last_movement)
+        """Return the last movement of the device (should be "Opening" or "Closing")"""
+        return self._last_movement
+
+    def last_position_set(self):
+        #_LOGGER.info("def last_position_set called=%s", self._last_position_set)
+        """Return the last position set of the device"""
+        return self._last_position_set
+        
+    @property
+    def last_command(self):
+        #_LOGGER.info("def last_command called=%s", self._last_command)
+        """Return the last command of the device"""
+        return self._last_command
+
+    @property
+    def is_open(self):
+        """Check if the cover is partially or fully open."""
+        #_LOGGER.info("def is_open called, self._current_cover_position = %s", self._current_cover_position)
+        if self._current_cover_position != 100:
+            #_LOGGER.info("is_open returning true)")
+            return True
+        #_LOGGER.info("is_open returning false")
+        return False
+
+    @property
+    def is_closed(self):
+        """Check if the cover is fully closed."""
+        #_LOGGER.info("def is_closed called")
+        if self._current_cover_position == 100:
+            #_LOGGER.info("is_closed returning true)")
+            return True
+        #_LOGGER.info("is_closed returning false")
+        return False
+
+    def update(self):
```

Comment:
You should not override `update` anymore. Rename this to `status_updated`. It will be called when an update has been performed and the "status" is available via `self._status` instead. Everything related to `self._available` can be removed as well, since exceptions are caught and handles availability automatically. You can look here if you want to know how it works:

https://github.com/rospogrigio/localtuya-homeassistant/blob/c266e0f90419aa52f0071a96d3258c022ebe9b51/custom_components/localtuya/__init__.py#L158